src/game/Spells/Scripts/Scripting/ClassScripts/Warlock.cpp

```diff
@@ -640,6 +640,29 @@ struct SeductionSuccubus : public AuraScript
     }
 };
 
+// 48181, 59161, 59163, 59164 - Haunt
+struct Haunt : public AuraScript
+{
+    enum
+    {
+        SPELL_HAUNT_HEAL = 48210,
+    };
+    void OnApply(Aura* aura, bool apply) const override
+    {
+        if (apply)
+        {
+            aura->GetModifier()->m_amount = 0;
+            return;
+        }
+        Unit* caster = aura->GetCaster();
+        Unit* target = aura->GetTarget();
```

Comment:
Target always exists, never need to worry about that in any scenario.

Reply:
I will keep stumbling over this, I fear. Do you see value in a non-nullable pointer type being added?

Reply:
Nope, gotta keep it in the noggin. In the best world it would be a reference, but I am not retroactively adding that. Add a comment to the function if that will make your life easier.